crates/forge/src/compatibility_check.rs

```diff
@@ -115,10 +117,80 @@ fn get_raw_version(name: &str, command: &RefCell<Command>) -> Result<String> {
         .to_string())
 }
 
+fn verify_contracts_sierra_versions(forge_config: &ForgeConfig) -> Result<()> {
+    if let Some(contract_names) = forge_config
+        .test_runner_config
+        .contracts_data
+        .get_contract_names()
+    {
+        for contract in contract_names {
+            let sierra_str = &forge_config
+                .test_runner_config
+                .contracts_data
+                .get_artifacts(contract)
+                .context(format!("Missing Sierra JSON for contract: {contract}"))?
+                .sierra;
+
+            let contract_version = parse_sierra_version(sierra_str)?;
+            if MINIMAL_SIERRA_VERSION_FOR_SIERRA_GAS > contract_version {
+                bail!(
+                    "Contract version {} is lower than required minimal sierra version",
+                    contract_version
+                );
+            }
+        }
+    }
+    Ok(())
+}
+
+fn parse_sierra_version(sierra_str: &str) -> Result<Version> {
+    let sierra_json: serde_json::Value =
+        serde_json::from_str(sierra_str).context("Failed to parse Sierra JSON")?;
+
+    let parsed_values: Vec<u8> = sierra_json["sierra_program"]
+        .as_array()
+        .context("Unable to read `sierra_program`. Ensure it is an array of felts.")?
```

Comment:
This is not user generated so it cannot be fixed by a user anyway.

```suggestion
        .context("Unable to read `sierra_program`.")?
```